### ISSUE TITLE
Bug: Audio dropouts/skips when windows open or screensaver kicks in (X86 only)

### ISSUE THREAD

**Issue by Gero5** [mpogue2/SquareDesk#810]
at the moment the auto player has little dropouts/skips in the playing music far too often. That happens when the screen saver or Lock Screen is started by MacOs (Intel) when changing windows or when the audio file list is updated.  it is very sensitive to system events. I think the continuous flow of music should have a high priority if not the highest because it is most important while calling.

**Comment by mpogue2** [comment]
On an older X86 Macbook, when the screensaver kicks in, there is a large spike in CPU utilization that apparently starves the PlayerThread.  Set Priority to TimeCriticalPriority, and the problem goes away.  It doesn't seem to use any more CPU cycles at all, on either M1 or X86 Macbooks.

NOTE: I was unable to recreate this problem on my M1 MacBook Air, no matter what I tried (opening multiple windows, moving them around, and invoking each of the system screensavers in turn).  However, the problem was pretty easy to reproduce (10 times out of 10) on my older MacBook Pro (X86, Big Sur), especially when a screensaver was started manually (move cursor into the upper left corner to start).  With window creation, audible interruptions were very intermittent and much harder to reproduce.

But, now that I had a reproducible test case, I was able to look into possible solutions.  Bumping the PlayerThread priority to TimeCriticalPriority made this problem go away entirely.  It did not increase CPU utilization at all, it just made sure that the PlayerThread got CPU time when it needed it.  I also looked for, but saw no negative effects on windows or screensavers, due to the PlayerThread getting higher priority than it did before.

FIXED by: d2a02c03d8d345ac8ae1b5349a72ffe87e9dc243

@Gero5 Thanks for the report!  I never noticed this before, because when using my older MacBook Pro during a dance, I never used a screensaver or created any new windows.  And, the problem was not reproducible on my M1 MacBook Air at all (the M1 has lots and lots of CPU cycles available, as well as twice as many CPU cores, so the PlayerThread always seems to get enough CPU cycles).
